pyccel/stdlib/cwrapper/cwrapper.c

```diff
@@ -0,0 +1,600 @@
+#include "cwrapper.h"
+
+
+
+// strings order needs to be the same as its equivalent numpy macro
+// https://numpy.org/doc/stable/reference/c-api/dtype.html
+const char* dataTypes[17] = {"Bool", "Int8", "UInt8", "Int16", "UIn16", "Int32", "UInt32",
+                             "Int64", "UInt64", "Int128", "UInt128", "Float32", "Float64",
+                             "Float128", "Complex64", "Complex128", "Complex256"};
+
+
+
+/*
+ * Functions : Cast functions
+ * --------------------------
+ * All functions listed down are based on C/python api
+ * with more tolerance to different precision
+ * Convert python type object to the desired C type
+ * Parameters :
+ *     object : the python object
+ * Returns    :
+ *     The desired C type, an error may be raised by c/python converter
+ *     so one should call PyErr_Occurred() to check for errors after the
+ *	   calling a cast function
+ * Reference of the used c python api function
+ * --------------------------------------------
+ * https://docs.python.org/3/c-api/float.html#c.PyFloat_AsDouble
+ * https://docs.python.org/3/c-api/complex.html#c.PyComplex_RealAsDouble
+ * https://docs.python.org/3/c-api/complex.html#c.PyComplex_ImagAsDouble
+ * https://docs.python.org/3/c-api/long.html#c.PyLong_AsLong
+ * https://docs.python.org/3/c-api/long.html#c.PyLong_AsLongLong
+ */
+
+float complex PyComplex_to_Complex64(PyObject *object)
+{
+	float complex	c;
+	float	real_part;
+	float	imag_part;
+
+	real_part = 0.0;
+	imag_part = 0.0;
+
+	// https://numpy.org/doc/1.17/reference/c-api.array.html#c.PyArray_IsScalar
+	// https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_ScalarAsCtype
+	if (PyArray_IsScalar(object, Complex64))
+		PyArray_ScalarAsCtype(object, &c);
+
+	else
+	{
+		real_part = (float)PyComplex_RealAsDouble(object);
+		imag_part = (float)PyComplex_ImagAsDouble(object);
+
+		c = CMPLXF(real_part, imag_part);
```

Comment:
Are you sure about this function? Windows doesn't seem to like it. If I google it I get mostly C++ references rather than C references. When translating expressions like this we usually write:
`real_part + imag_part * _Complex_I`
I found a reference for the contents of `complex.h` [here](https://pubs.opengroup.org/onlinepubs/007904975/basedefs/complex.h.html), it doesn't mention `cmplxf`, and `cmplx`. Maybe they are extensions available with some compilers but not in the standard.
To be sure I think we should use the same syntax as normally with `_Complex_I`

Reply:
According to [this](https://en.cppreference.com/w/c/numeric/complex/CMPLX) they are available from C11. Normally we stick to C99